src/components/overview/MacroList.tsx

```diff
@@ -1,27 +1,68 @@
 import { AddIcon } from '@chakra-ui/icons'
 import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
   Box,
   Button,
   Flex,
   Grid,
   GridItem,
+  HStack,
+  Text,
   useColorModeValue,
   VStack
 } from '@chakra-ui/react'
-import { useCallback } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 import { useApplicationContext } from '../../contexts/applicationContext'
 import { useSelectedCollection } from '../../contexts/selectors'
 import { ViewState } from '../../constants/enums'
-import { Macro } from '../../types'
+import { Collection, Macro } from '../../types'
 import MacroCard from './MacroCard'
 import useScrollbarStyles from '../../hooks/useScrollbarStyles'
 import useMainBgColour from '../../hooks/useMainBgColour'
+import { motion } from 'framer-motion'
+
+import { borderRadiusStandard } from '../../theme/config'
 
 export default function MacroList() {
-  const { selection, onCollectionUpdate, changeViewState } =
-    useApplicationContext()
+  // Get the ApplicationContext from your hook.
+  const {
+    collections,
+    selection,
+    onCollectionUpdate,
+    changeViewState,
+    searchValue,
+    isMacroOutputEnabled
+  } = useApplicationContext()
   const currentCollection = useSelectedCollection()
   const shadowColour = useColorModeValue('md', 'white-md')
+  const [matchingMacros, setMatchingMacros] = useState<
+    { macro: Macro; collection: Collection }[]
+  >([])
+
+  useEffect(() => {
+    const newMatchingMacros: { macro: Macro; collection: Collection }[] = []
+
+    if (searchValue) {
+      collections.map((collection) => {
+        collection.macros.map((macro) => {
+          if (macro.name.toLocaleLowerCase().includes(searchValue)) {
+            newMatchingMacros.push({ macro, collection })
+          }
+        })
+      })
+      setMatchingMacros(newMatchingMacros)
+    } else {
+      setMatchingMacros(
+        currentCollection.macros.map((macro, index) => ({
+          macro,
+          collection: currentCollection,
+          index
+        }))
+      )
+    }
+  }, [searchValue, collections, currentCollection])
```

Comment:
Search macro logic

Reply:
Couldn't this be a useMemo rather than a useEffect?

Reply:
Indeed it could, swapped: [54b1f21](https://github.com/WootingKb/wooting-macros/pull/194/commits/54b1f218c95ea16cfa77372536d65d78e6e4e817)